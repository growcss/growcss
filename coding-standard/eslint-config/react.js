module.exports ={
  extends: [
    'prettier/react',
    'plugin:jest/recommended'
  ],
  plugins: [
    'react',
    'jsx-a11y',
    // 'security', @todo check it on the next version
  ],
  rules: {
    ...{
      'react/sort-comp': 'off',
      'react/jsx-filename-extension': 'off',
      'react/require-default-props': 'off',
    },
    ...require('./rules/prettier-conflicts-react'),
    ...require('./rules/jest'),
  },
  env: {
    browser: true,
    commonjs: true,
    es6: true,
    jest: true,
    node: true,
  },
};
