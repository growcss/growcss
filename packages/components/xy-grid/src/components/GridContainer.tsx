import React, { Component } from 'react';
import classNames from 'classnames';
import remCalc from '@growcss/util-remcalc';
import { Gutters as DefaultGutters, GuttersProps } from './Gutters';
import { GridContainerElement } from '../styled/GridContainerElement';
import { GridProps } from './AbstractGrid';

export type GridContainerProps = {
    children: React.Component<GridProps>[],
    type?: string,
    width?: string | number,
    gutterSizes?: string | number | GuttersProps,
}

type GridContainerDefaultProps = {
  width: string | number,
  gutterSizes: GuttersProps
}

export default class GridContainer extends Component<GridContainerProps> {
  static defaultProps: GridContainerDefaultProps = {
    width: remCalc(1200),
    gutterSizes: DefaultGutters,
  };

  render() {
    const {
      type,
      children,
      width,
      gutterSizes,
      ...other
    } = this.props;
    const className = classNames('gc-grid-container');

    let maxWidth = width;
    let gutter = gutterSizes;

    if (type === 'fluid' || type === 'full') {
      maxWidth = '100%';
    }

    if (type === 'full') {
      gutter = '0';
    }

    return (
      <GridContainerElement
        className={className}
        maxWidth={maxWidth!}
        gutterSizes={gutter!}
        type={type}
        {...other}
      >
        {children}
      </GridContainerElement>
    );
  }
}
