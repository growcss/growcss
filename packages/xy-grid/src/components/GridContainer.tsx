import * as React from 'react';
import classNames from 'classnames';
import {rem} from '@growcss/elaborate';
import {Gutters as DefaultGutters,GuttersProps} from './Gutters';
import {GridContainerElement} from '../styled/GridContainerElement';

export interface GridContainerProps extends React.HTMLAttributes<HTMLDivElement> {
  children: React.ReactNode,
  type?: string,
  width?: string | number,
  gutterSizes?: string | number | GuttersProps,
}

type GridContainerDefaultProps = {
  width: string | number,
  gutterSizes: GuttersProps,
}

export default class GridContainer extends React.Component<GridContainerProps> {
  static defaultProps: GridContainerDefaultProps = {
    width: rem(1200),
    gutterSizes: DefaultGutters,
  };

  render() {
    const {
      type,
      children,
      width,
      gutterSizes,
      ...other
    } = this.props;
    const className = classNames('gc-grid-container');

    let maxWidth = width;
    let gutter = gutterSizes;

    if (type === 'fluid' || type === 'full') {
      maxWidth = '100%';
    }

    if (type === 'full') {
      gutter = '0';
    }

    return (
      <GridContainerElement
        className={className}
        maxWidth={maxWidth!}
        gutterSizes={gutter!}
        type={type}
        {...other}
      >
        {children}
      </GridContainerElement>
    );
  }
}
