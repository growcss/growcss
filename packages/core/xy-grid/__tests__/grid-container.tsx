import * as React from 'react';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import GridContainer from '../src/components/grid-container';

test('if the container is centered and have a max-width of 75rem', () => {
  const container = <GridContainer>test</GridContainer>;
  const tree = renderer.create(container).toJSON();

  expect(shallow(container)).toMatchSnapshot();
  expect(tree).toHaveStyleRule('max-width', '75rem');
  expect(tree).toHaveStyleRule('margin', '0 auto');
});

test('if the container is centered and has a max-width of 100%', () => {
  const container = <GridContainer type="fluid">test</GridContainer>;
  const tree = renderer.create(container).toJSON();

  expect(shallow(container)).toMatchSnapshot();
  expect(tree).toHaveStyleRule('max-width', '100%');
  expect(tree).toHaveStyleRule('margin', '0 auto');
});

test('if the container content is of the full width of the available space', () => {
  const container = <GridContainer type="full">test</GridContainer>;
  const tree = renderer.create(container).toJSON();

  expect(shallow(container)).toMatchSnapshot();
  expect(tree).toHaveStyleRule('max-width', '100%');
  expect(tree).toHaveStyleRule('overflow-x', 'hidden');
});
