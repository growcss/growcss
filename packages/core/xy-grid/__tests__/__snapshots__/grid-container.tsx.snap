// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if the container content is of the full width of the available space 1`] = `
<grid-container-element__GridContainerElement
  className="gc-grid-container"
  gutterSizes="0"
  maxWidth="100%"
  type="full"
>
  test
</grid-container-element__GridContainerElement>
`;

exports[`if the container is centered and has a max-width of 100% 1`] = `
<grid-container-element__GridContainerElement
  className="gc-grid-container"
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  maxWidth="100%"
  type="fluid"
>
  test
</grid-container-element__GridContainerElement>
`;

exports[`if the container is centered and have a max-width of 75rem 1`] = `
<grid-container-element__GridContainerElement
  className="gc-grid-container"
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  maxWidth="75rem"
>
  test
</grid-container-element__GridContainerElement>
`;
